### PR TITLE
[Feature][dinky-geteway] Feature Obtain job information using the ingress address #3871

### PR DESCRIPTION
**Pull request by jianjun159**
## Purpose of the pull request

Feature Obtain job information using the ingress address
image

## Brief change log

dinky-getaway moudle add ingress support
Be related to issues #3873 

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by aiwenmo** [comment]
The configuration of the ingress address should follow the cluster configuration of k8s.

**Comment by jianjun159** [comment]
> ingress地址的配置需要遵循k8s的集群配置。

I don't think it is necessary to bind to the K8s cluster, because in actual use, you can use the same ingress domain address even if you use multiple clusters

**Comment by aiwenmo** [comment]
Is there a situation where multiple clusters use different addresses? I think we can first implement the situation you mentioned, and then consider the situation I mentioned later. Thank you very much for your contribution.

File: dinky-common/src/main/resources/i18n/messages_zh_CN.properties [hunk 1]
```diff
@@ -325,3 +325,8 @@ sys.flink.settings.flinkHistoryServerPort=Flink History Server 端口
 sys.flink.settings.flinkHistoryServerPort.note=Flink History Server 端口，例如：8082，确保端口没有被占用
 sys.flink.settings.flinkHistoryServerArchiveRefreshInterval= Flink History Server 刷新间隔
 sys.flink.settings.flinkHistoryServerArchiveRefreshInterval.note=Flink History Server 刷新间隔，单位：毫秒，例如：10000，表示每隔10秒刷新一次
+
+sys.ingress.settings.enable=是否启用Ingress
+sys.ingress.settings.enable.note=启用Ingress后，在Kubernetes Application模式下，通过Ingress地址获取Job运行状
```
Comment (aiwenmo) [inline dinky-common/src/main/resources/i18n/messages_zh_CN.properties]: sys.ingress.settings.enable.note=启用Ingress后，在Kubernetes Application模式下，通过Ingress地址获取Job运行状态

File: dinky-gateway/src/main/java/org/dinky/gateway/kubernetes/KubernetesApplicationGateway.java [hunk 2]
```diff
@@ -87,7 +101,24 @@ public GatewayResult submitJar(FlinkUdfPathContextHolder udfPathContextHolder) {
...
-            KubernetesResult kubernetesResult = waitForJmAndJobStart(kubernetesClient, deployment, clusterClient);
+            KubernetesResult kubernetesResult;
+
+            boolean ingressEnable = SystemConfiguration.getInstances().getIngressEnable();
+            String ingressDomain = SystemConfiguration.getInstances().getIngressDomain();
+            // if ingress is enabled and ingress domain is not empty, create an ingress service
+            if (ingressEnable && StringUtils.isNotEmpty(ingressDomain)) {
+                K8sClientHelper k8sClientHelper = getK8sClientHelper();
+                long ingressStart = SystemClock.now();
+                DinkyKubernetesIngress ingress = new DinkyKubernetesIngress(getK8sClientHelper());
```
Comment (aiwenmo) [inline dinky-gateway/src/main/java/org/dinky/gateway/kubernetes/KubernetesApplicationGateway.java]: DinkyKubernetesIngress ingress = new DinkyKubernetesIngress(k8sClientHelper);

**Comment by Zzm0809** [comment]
Thank you for your contribution!! 
It is recommended to add this configuration in `cluster configuration` when selecting `k8s` related mode. This configuration should be 1v1 compared to the cluster configuration!

**Comment by jianjun159** [comment]
> Thank you for your contribution!! It is recommended to add this configuration in `cluster configuration` when selecting `k8s` related mode. This configuration should be 1v1 compared to the cluster configuration!

First of all, thank you for your reply. I think your suggestion is correct. I will modify the code according to your suggestion, verify it, and submit the code again

File: dinky-gateway/src/main/java/org/dinky/gateway/kubernetes/KubernetesApplicationGateway.java (hunk 2 above)
Comment (jianjun159) [reply dinky-gateway/src/main/java/org/dinky/gateway/kubernetes/KubernetesApplicationGateway.java]: Thank you for your suggestion. I will revise this part of the code again, verify it and submit it again

**Comment by jianjun159** [comment]
> Thank you for your contribution!! It is recommended to add this configuration in `cluster configuration` when selecting `k8s` related mode. This configuration should be 1v1 compared to the cluster configuration!

企业微信截图_cdb822b7-643b-40a7-9a22-aed7385aaade
It is currently bound to the cluster

**Comment by jianjun159** [comment]
@aiwenmo @Zzm0809 Hello Dear, Follow up your proposal, now it is modified to bind with the cluster, and I have submitted the relevant front and back end code, I hope you two bosses can help review it, thank you

File: dinky-web/src/pages/RegCenter/Cluster/Configuration/components/ConfigurationModal/ConfigurationForm/FlinkK8s/index.tsx [hunk 3]
```diff
@@ -152,6 +153,12 @@ const FlinkK8s = (props: { type: string; value: any; form: FormInstance<Values>
     }
   ];
 
+  const ingressEnabled = Form.useWatch(
+    ['config', 'kubernetesConfig', 'ingressConfig', 'kubernetes.ingress.enabled'],
+    form
+  );
+  console.log(ingressEnabled);
```
Comment (Zzm0809) [inline dinky-web/src/pages/RegCenter/Cluster/Configuration/components/ConfigurationModal/ConfigurationForm/FlinkK8s/index.tsx]: Please delete this line

File: dinky-web/src/pages/SettingCenter/GlobalSetting/SettingOverView/index.tsx [hunk 4]
```diff
@@ -104,7 +104,8 @@ const SettingOverView = () => {
       dolphinscheduler: dsConfig,
       ldap: ldapConfig,
       metrics: metricsConfig,
-      resource: resourceConfig
+      resource: resourceConfig,
+      ingress: ingressConfig
```
Comment (Zzm0809) [inline dinky-web/src/pages/SettingCenter/GlobalSetting/SettingOverView/index.tsx]: Delete this line

Comment (jianjun159) [reply dinky-web/src/pages/SettingCenter/GlobalSetting/SettingOverView/index.tsx]: It's been deleted

File: dinky-web/src/pages/RegCenter/Cluster/Configuration/components/ConfigurationModal/ConfigurationForm/FlinkK8s/index.tsx (hunk 3 above)
Comment (jianjun159) [reply dinky-web/src/pages/RegCenter/Cluster/Configuration/components/ConfigurationModal/ConfigurationForm/FlinkK8s/index.tsx]: It's been deleted,

**Review by Zzm0809 — APPROVED** [review]
LGTM

**Pull request merged**